File: MyProject/credito.php
<?php
// ejemplo.php
if (isset($argv[1])) {
    $parametro = $argv[2];
    echo "El parámetro recibido es: $parametro";
} else {
    echo "No se recibió ningún parámetro.";
}


function encryptCedula($cedula)
{
    // Contenido de la clave pública
    $public_key_file = dirname(__DIR__) . "/models/PBKey.txt";
    // Lee el contenido del archivo PEM
    $public_key_content = file_get_contents($public_key_file);
    // Elimina espacios en blanco adicionales alrededor del contenido
    $public_key_content = trim($public_key_content);

    $rsaKey = openssl_pkey_get_public($public_key_content);
    if (!$rsaKey) {
        // Manejar el error de obtener la clave pública
        return [0, openssl_error_string(), $public_key_file];
    }
    // // Divide el texto en bloques para encriptar
    $encryptedData = '';
    $encryptionSuccess = openssl_public_encrypt($cedula, $encryptedData, $rsaKey);

    // Obtener detalles del error, si hubo alguno
    // $error = openssl_error_string();
    // if ($error) {
    //     // Manejar el error de OpenSSL
    //     return $error;
    // }

    // Liberar la clave pública RSA de la memoria
    openssl_free_key($rsaKey);

    if ($encryptionSuccess === false) {
        // Manejar el error de encriptación
        return [0, null, $public_key_file];
    }

    // Devolver la cédula encriptada
    return [1, base64_encode($encryptedData)];
    // echo json_encode(base64_encode($encryptedData));
    // exit();
    // return ($encrypted);
}

function Obtener_Datos_Credito($param, $param_DATOS, $val)
{
    try {
        // $old_error_reporting = error_reporting();
        // Desactivar los mensajes de advertencia
        // error_reporting($old_error_reporting & ~E_WARNING);
        if ($val == 1) {
            $cedula = $param->CEDULA;
            $nacimiento = $param->FECHA_NACIM;
            $CELULAR = base64_decode($param_DATOS["celular"]);
        } else {
            $cedula = $param["CEDULA"];
            $nacimiento = $param["FECHA_NACIM"];
            $CELULAR = ($param_DATOS["celular"]);
        }



        // $cedula = "0930254909";
        $cedula_ECrip = $this->encryptCedula($cedula);
        if ($cedula_ECrip[0] == 0) {
            return [0, $cedula_ECrip, [], []];
        } else {
            $cedula_ECrip = $cedula_ECrip[1];
        }

        $fecha = DateTime::createFromFormat('d/m/Y', $nacimiento);
        $fecha_formateada = $fecha->format('Ymd');
        $ingresos = "500";
        $Instruccion = "SECU";
        $SEC = $this->Get_Secuencial_Api_Banco();
        $SEC = intval($SEC[0]["valor"]) + 1;

        $data = array(
            "transaccion" => 4001,
            "idSession" => "1",
            "secuencial" => $SEC,
            "mensaje" => array(
                "IdCasaComercialProducto" => 8,
                "TipoIdentificacion" => "CED",
                "IdentificacionCliente" => $cedula_ECrip, // Encriptar la cédula
                "FechaNacimiento" => $fecha_formateada,
                "ValorIngreso" => $ingresos,
                "Instruccion" =>  $Instruccion,
                "Celular" =>  $CELULAR
            )
        );

        // echo json_encode($data);
        // exit();
        // Convertir datos a JSON
        $data_string = json_encode($data);
        // URL del API
        $url = 'https://bs-autentica.com/cco/apiofertaccoqa1/api/CasasComerciales/GenerarCalificacionEnPuntaCasasComerciales';
        // API Key
        $api_key = '0G4uZTt8yVlhd33qfCn5sazR5rDgolqH64kUYiVM5rcuQbOFhQEADhMRHqumswphGtHt1yhptsg0zyxWibbYmjJOOTstDwBfPjkeuh6RITv32fnY8UxhU9j5tiXFrgVz';
        // Inicializa la sesión cURL
        $ch = curl_init($url);

        curl_setopt($ch, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_V4);
        // Configura las opciones de la solicitud
        $verbose = fopen('php://temp', 'w+');
        curl_setopt($ch, CURLOPT_VERBOSE, true);
        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "PUT");
        curl_setopt($ch, CURLOPT_POSTFIELDS, $data_string);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        curl_setopt($ch, CURLOPT_HTTPHEADER, array(
            'Content-Type: application/json',
            'Content-Length: ' . strlen($data_string),
            'ApiKeySuscripcion: ' . $api_key
        ));
        curl_setopt($ch, CURLOPT_USERAGENT, 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.7; rv:7.0.1) Gecko/20100101 Firefox/7.0.1');

        // Ejecuta la solicitud y obtiene la respuesta
        $response = (curl_exec($ch));
        // Cierra la sesión cURL
        $error = (curl_error($ch));
        curl_close($ch);
        // Imprime la respuesta
        // echo $response;
        // return [1, $ARRAY];
        rewind($verbose);
        $verboseLog = stream_get_contents($verbose);
        $response_array = json_decode($response, true);

        $this->Update_Secuencial_Api_Banco($SEC);

        // echo json_encode($response_array);
        // exit();
        // if (extension_loaded('curl')) {
        //     echo "cURL está habilitado en este servidor.";
        // } else {
        //     echo "cURL no está habilitado en este servidor.";
        // }

        // Verificar si hay un error en la respuesta
        if (isset($response_array['esError'])) {

            $_inci = array(
                "ERROR_TYPE" => "API_SOL",
                "ERROR_CODE" => $response_array['codigo'],
                "ERROR_TEXT" => $response_array['esError'] . "-"
                    . $response_array['descripcion'] . "-"
                    . $response_array['idSesion'] . "-"
                    . $response_array['secuencial'],
            );
            date_default_timezone_set('America/Guayaquil');
            $hora_actual = date('G');

            if ($response_array['esError'] == true) {
                if ($response_array['descripcion'] == "No tiene oferta") {
                    $INC = $this->INCIDENCIAS($_inci);
                    return [2, $response_array, $data, $INC];
                } else if ($response_array['descripcion'] == "Ha ocurrido un error" && $hora_actual >= 21) {
                    $INC = $this->INCIDENCIAS($_inci);
                    return [3, $response_array, $data, $INC, $hora_actual];
                }
            } else {
                $INC = $this->INCIDENCIAS($_inci);
                return [1, $response_array, $data];
            }
        } else {
            // $INC = $this->INCIDENCIAS($_inci);

            return [0, $response_array, $data, $error, $verboseLog, extension_loaded('curl')];
        }
    } catch (Exception $e) {
        // Captura la excepción y maneja el error
        // echo "Error: " . $e->getMessage();
        $param = array(
            "ERROR_TYPE" => "API_SOL_FUNCTION",
            "ERROR_CODE" => "",
            "ERROR_TEXT" => $e->getMessage(),
        );
        $this->INCIDENCIAS($param);
        return [0, "Error al procesar la solictud banco", $e->getMessage()];
    }
}

File: MyProject/credito.py
from asyncio import sleep
import subprocess
import time
import mysql.connector
from mysql.connector import Error
import json
import base64
# import requests
from datetime import datetime
# from Crypto.PublicKey import RSA
# from Crypto.Cipher import PKCS1_OAEP
import os
import subprocess


# //pip install pycryptodome


def Cargar_Datos():
     
    while True:
        try:
            time.sleep(0.4)
            conexion = mysql.connector.connect(
                host="50.87.184.179",
                user="wsoqajmy_jorge",
                password="Equilivre3*",
                database="wsoqajmy_crediweb"
            )
            # conexion = mysql.connector.connect(
            #     host="localhost",
            #     user="root",
            #     password="",
            #     database="crediweb"
            # )
            if conexion.is_connected():
                # print("Conexión establecida")
                pass

            cursor = conexion.cursor()
            consulta = 'SELECT * FROM creditos_solicitados WHERE estado = 1 and EST_REGISTRO = 1'
            # valores = (numero,)
            cursor.execute(consulta)
            resultados = cursor.fetchall()
            print(len(resultados))
            for row in resultados:
                print(row[2])
                cedula = row[2]
                fecha = row[6]
                celular = row[3]
                ID_UNICO = row[1]
                parametro = [cedula,fecha,celular,ID_UNICO]
                ruta_php = "C:/xampp/php/php.exe"
                ruta = "C:\\xampp\\htdocs\\credito_express_api\\credito.php"
                comando = [ruta_php, ruta] + parametro
                resultado = subprocess.run(comando, capture_output=True, text=True)
                print(resultado.stdout)


            cursor.close()
            conexion.close()

        except Error as e:
            print("Error de conexión:", e)
            print("Intentando reconectar...")
            continue

# def get_secuencial_api_banco():
#     try:
#             conexion = mysql.connector.connect(
#                 host="50.87.184.179",
#                 user="wsoqajmy_jorge",
#                 password="Equilivre3*",
#                 database="wsoqajmy_crediweb"
#             )
#             if conexion.is_connected():
#                 # print("Conexión establecida")
#                 pass

#             cursor = conexion.cursor()
#             consulta = 'SELECT * FROM parametros WHERE id = 1'
#             # valores = (numero,)
#             cursor.execute(consulta)
#             resultados = cursor.fetchall()
#             for row in resultados:
#                 # print(row[2])
#                 return row[2]
#             cursor.close()
#             conexion.close()

#     except Exception as e:
#             return [0, "INTENTE DE NUEVO"]

# def UPDATE_secuencial_api_banco(sec):
#         try:
#             # conexion = mysql.connector.connect(
#             #     host="localhost",
#             #     user="root",
#             #     password="",
#             #     database="crediweb"
#             # )
#             conexion = mysql.connector.connect(
#                 host="50.87.184.179",
#                 user="wsoqajmy_jorge",
#                 password="Equilivre3*",
#                 database="wsoqajmy_crediweb"
#             )

#             cursor = conexion.cursor()
#             consulta = """
#                 UPDATE parametros
#                     set valor = %s
#                 WHERE id = 1
#             """
#             valores = (
#                  sec,
#             )
#             cursor.execute(consulta,valores)
#             conexion.commit()

#         except Error as e:
#             print("Error de conexión:", e)
         
# def encrypt_cedula(cedula):
#   # Obtener la ruta completa del archivo de clave pública (PBKey.txt)
#     public_key_file = os.path.join(os.path.dirname(__file__), 'PBKey.txt')

#     try:
#         # Leer la clave pública desde el archivo
#         with open(public_key_file, 'rb') as f:
#             key = RSA.import_key(f.read())

#         # Crear un objeto Cipher con la clave pública
#         cipher = PKCS1_OAEP.new(key)

#         # Encriptar la cédula
#         encrypted_data = cipher.encrypt(cedula.encode('utf-8'))

#         # Devolver la cédula encriptada en formato base64
#         return base64.b64encode(encrypted_data).decode('utf-8')
#     except Exception as e:
#         # Manejar el error de encriptación
#         return (0, str(e), public_key_file)

# def obtener_datos_credito():
#     try:
#         # if val == 1:
#         #     cedula = param["CEDULA"]
#         #     nacimiento = param["FECHA_NACIM"]
#         #     celular = base64.b64decode(param_datos["celular"])
#         # else:
#         #     cedula = param["CEDULA"]
#         #     nacimiento = param["FECHA_NACIM"]
#         #     celular = param_datos["celular"]
#         cedula = "0951991637"
#         nacimiento = "19960114"
#         celular = "0969786231"
#         # fecha = datetime.strptime(nacimiento, '%d/%m/%Y')
#         # fecha_formateada = fecha.strftime('%Y%m%d')
#         ingresos = "1500"
#         instruccion = "SECU"
#         sec = get_secuencial_api_banco()
#         sec = int(sec) + 1
#         UPDATE_secuencial_api_banco(sec)

#         data = {
#             "transaccion": 4001,
#             "idSession": "1",
#             "secuencial": sec,
#             "mensaje": {
#                 "IdCasaComercialProducto": 8,
#                 "TipoIdentificacion": "CED",
#                 "IdentificacionCliente": encrypt_cedula(cedula),
#                 "FechaNacimiento": nacimiento,
#                 "ValorIngreso": ingresos,
#                 "Instruccion": instruccion,
#                 "Celular": celular
#             }
#         }

#         url = 'https://bs-autentica.com/cco/apiofertaccoqa1/api/CasasComerciales/GenerarCalificacionEnPuntaCasasComerciales'
#         api_key = '0G4uZTt8yVlhd33qfCn5sazR5rDgolqH64kUYiVM5rcuQbOFhQEADhMRHqumswphGtHt1yhptsg0zyxWibbYmjJOOTstDwBfPjkeuh6RITv32fnY8UxhU9j5tiXFrgVz'

#         headers = {
#             'Content-Type': 'application/json',
#             'ApiKeySuscripcion': api_key
#         }

#         response = requests.put(url, json=data, headers=headers, verify=False)
#         response_json = response.json()

#         print(response_json)
#         # update_secuencial_api_banco(sec)

#         # if 'esError' in response_json:
#         #     if response_json['esError']:
#         #         return (0, response_json, data)
#         #     elif response_json['descripcion'] == "No tiene oferta":
#         #         return (2, response_json, data)
#         #     elif response_json['descripcion'] == "Ha ocurrido un error" and datetime.now().hour >= 21:
#         #         return (3, response_json, data, datetime.now().hour)
#         #     else:
#         #         return (1, response_json, data)
#         # else:
#         #     return (0, response_json, data, response.content, response.text, 'curl' in dir())

#     except Exception as e:
#         param = {
#             "ERROR_TYPE": "API_SOL_FUNCTION",
#             "ERROR_CODE": "",
#             "ERROR_TEXT": str(e),
#         }
#         # incidencias(param)
#         return (0, "Error al procesar la solicitud del banco", str(e))


if __name__ == "__main__":
    Cargar_Datos()